src/PixiEditor.ChangeableDocument/Changeables/Graph/Context/FuncContext.cs

```diff
@@ -98,17 +99,58 @@ public Half4 NewHalf4(Expression r, Expression g, Expression b, Expression a)
         if (!HasContext && r is Float1 firstFloat && g is Float1 secondFloat && b is Float1 thirdFloat && a is Float1 fourthFloat)
         {
             Half4 constantHalf4 = new Half4("");
-            byte rByte = (byte)firstFloat.ConstantValue;
-            byte gByte = (byte)secondFloat.ConstantValue;
-            byte bByte = (byte)thirdFloat.ConstantValue;
-            byte aByte = (byte)fourthFloat.ConstantValue;
+            byte rByte = firstFloat.FullSizeByteConstant;
+            byte gByte = secondFloat.FullSizeByteConstant;
+            byte bByte = thirdFloat.FullSizeByteConstant;
+            byte aByte = fourthFloat.FullSizeByteConstant;
             constantHalf4.ConstantValue = new Color(rByte, gByte, bByte, aByte);
             return constantHalf4;
         }
         
         return Builder.ConstructHalf4(r, g, b, a);
     }
+    
+    public Half4 HsvaToRgba(Expression h, Expression s, Expression v, Expression a)
+    {
+        if (!HasContext && h is Float1 firstFloat && s is Float1 secondFloat && v is Float1 thirdFloat && a is Float1 fourthFloat)
+        {
+            Half4 constantHalf4 = new Half4("");
+            var hValue = firstFloat.ConstantValue * 360;
+            var sValue = secondFloat.ConstantValue * 100;
+            var vValue = thirdFloat.ConstantValue * 100;
+            byte aByte = fourthFloat.FullSizeByteConstant;
+            constantHalf4.ConstantValue = Color.FromHsv((float)hValue, (float)sValue, (float)vValue, aByte);
+            return constantHalf4;
+        }
+
+        if (Half4Float1Accessor.AllAccessSame(h, s, v, a, out var half4))
+        {
+            return Builder.AssignNewHalf4(Builder.Functions.GetHsvToRgb(half4));
+        }
+        
+        return Builder.AssignNewHalf4(Builder.Functions.GetHsvToRgb(h, s, v, a));
```

Comment:
So if `h`, `s`, `v`, `a` components are part of the same half4, function produces new half4 based on the parent half4 instead of child components? What advantage does it have over just passing the components itself (second return)?